products/sentinel_one.py

```diff
@@ -479,89 +526,179 @@ def _process_queries(self):
+                "toDate": datetime_to_epoch_millis(end_date),
+                "limit": 20000,
+                "query": merged_query
+            })
+
+            if not self._pq:
                 params.update({
-                    "fromDate": datetime_to_epoch_millis(start_date),
                     "isVerbose": False,
                     "queryType": ['events'],  # options: 'events', 'procesState'
-                    "limit": 20000,
-                    "toDate": datetime_to_epoch_millis(end_date),
-                    "query": merged_query
                 })
 
-                self.log.debug(f'Query params: {params}')
+            self.log.debug(f'Query params: {params}')
 
+            if not self._pq:
                 # ensure we do not submit more than one request every 60 seconds to comply with rate limit
                 seconds_sice_last_request = time.time() - self._last_request
                 if seconds_sice_last_request < 60:
                     sleep_seconds = 60 - seconds_sice_last_request
                     self.log.debug(f'Sleeping for {sleep_seconds}')
-                    time.sleep(sleep_seconds)
 
-                # start deep visibility API call
-                query_response = self._session.post(self._build_url('/web/api/v2.1/dv/init-query'),
-                                                    headers=self._get_default_header(), data=json.dumps(params))
-                self._last_request = time.time()
+                    cancel_event.wait(ceil(sleep_seconds))
+
+            # start deep visibility API call
+            url = '/web/api/v2.1/dv/events/pq' if self._pq else '/web/api/v2.1/dv/init-query'
+            query_response = self._session.post(self._build_url(url),
+                                                headers=self._get_default_header(), data=json.dumps(params))
+            self._last_request = time.time()
 
-                body = query_response.json()
-                if 'errors' in body and any(('could not parse query' in x['detail'] for x in body['errors'])):
-                    raise ValueError(f'S1 could not parse query "{merged_query}"')
+            body = query_response.json()
+            if 'errors' in body and any(('could not parse query' in x['detail'] for x in body['errors'])):
+                raise ValueError(f'S1 could not parse query "{merged_query}"')
 
-                self.log.debug(query_response.json())
-                query_response.raise_for_status()
+            self.log.debug(query_response.json())
+            query_response.raise_for_status()
 
-                query_id = body['data']['queryId']
-                self.log.info(f'Query ID is {query_id}')
+            query_id = body['data']['queryId']
+            self.log.info(f'Query ID is {query_id}')
 
-                events = self._get_dv_events(query_id)
-                self.log.debug(f'Got {len(events)} events')
+            events = self._get_dv_events(query_id, p_bar=p_bar, cancel_event=cancel_event)
```

Comment:
When testing, I came across this issue when the script checked the status of a PowerQuery, 503 errors got thrown regardless of if the query is valid or if there's results being returned. Adding this section appeared to clear up the errors. I'm wondering if the API doesn't allow you to ping a PowerQuery's status if the search is finished in the initial call?

```suggestion
            if self._pq and body['data']['status'] == 'FINISHED': # If using PQ, the results can be returned immediately
                events = body['data']['data']
            else:
                events = self._get_dv_events(query_id, p_bar=p_bar, cancel_event=cancel_event)
```